src/lib.rs

```diff
@@ -126,7 +126,7 @@ fn is_prohibited_bidirectional_text(s: &str) -> bool {
 pub fn nameprep(s: &str) -> Result<Cow<'_, str>, Error> {
     // fast path for ascii text
     if s.chars()
-        .all(|c| c.is_ascii_lowercase() && !tables::ascii_control_character(c))
+        .all(|c| c.is_ascii() && !c.is_ascii_uppercase() && !tables::ascii_control_character(c))
```

Comment:
It looks like we also need to reject ascii spaces, which are prohibited.

Reply:
I have simplified this expression, as ASCII domains only allow letters, digits, '-' and '.' actually.